akka-persistence-typed/src/main/scala/akka/persistence/typed/state/scaladsl/Effect.scala

```diff
@@ -25,7 +25,12 @@ object Effect {
    */
   def persist[State](state: State): EffectBuilder[State] = Persist(state)
 
-  // FIXME add delete effect
+  /**
+   * Reset to the initial state and delete the persisted state.
+   *
+   * Side effects can be chained with `thenRun`
+   */
+  def reset[State](): EffectBuilder[State] = Reset()
```

Comment:
I'm wondering if it's not better to call it `delete` instead. The fact that a reset happens is just a consequence of the deletion and so don't need to be explicit in the API. 

What I mean is that if we delete a row, we can always add it again assuming the primary key is not an auto-increment one. If we send a new create command for that same entity id, we will by design get the initial state and it will behave as if it was the first time that we call it. 

If I don't want a deletion, but real reset. Then I can simply have a command that ignores the existing state and pass the initial state to `persist`. So, `reset` is easily implementable by the users. 

Also, an explicit deletion will fit better with the query issue raised by Patrik in his comment https://github.com/akka/akka/pull/31071/files#r788886575